qutip/solver/mcsolve.py

```diff
@@ -0,0 +1,311 @@
+        is passed in options (per default, mcsolve will stop after 1e8 sec)::
+            ``options.mcsolve['map_options']['timeout'] = max_sec``
+        Or if the target tolerance is reached, see ``target_tol``.
+
+    args : dict, [optional]
+        Arguments for time-dependent Hamiltonian and collapse operator terms.
+
+    options : SolverOptions, [optional]
+        Options for the evolution.
+
+    seeds : int, SeedSequence, list, [optional]
+        Seed for the random number generator. It can be a single seed used to
+        spawn seeds for each trajectory or a list of seeds, one for each
+        trajectory. Seeds are saved in the result and they can be reused with::
+            seeds=prev_result.seeds
+
+    target_tol : float, list, [optional]
+        Target tolerance of the evolution. The evolution will compute
+        trajectories until the error on the expectation values is lower than
+        this tolerance. The error is computed using jackknife resampling.
+        ``target_tol`` can be an absolute tolerance or a pair of absolute and
+        relative tolerance, in that order. Lastly, it can be a list of pairs of
+        (atol, rtol) for each e_ops.
+
+    timeout : float [optional]
+        Maximum time for the evolution in second. When reached, no more
+        trajectories will be computed. Overwrite the option of the same name.
+
+    Returns
+    -------
+    results : :class:`qutip.solver.Result`
+        Object storing all results from the simulation. Which results is saved
+        depends on the presence of ``e_ops`` and the options used. ``collapse``
+        and ``photocurrent`` is available to Monte Carlo simulation results.
+    """
+    H = QobjEvo(H, args=args, tlist=tlist)
+    if not isinstance(c_ops, (list, tuple)):
+        c_ops = [c_ops]
+    c_ops = [QobjEvo(c_op, args=args, tlist=tlist) for c_op in c_ops]
+
+    if len(c_ops) == 0:
+        return mesolve(H, state, tlist, e_ops=e_ops, args=args,
+                       options=options)
+
+    if isinstance(ntraj, list):
+        raise TypeError("No longer supported, use `result.expect_traj_avg`"
+                        "with the options `keep_runs_results=True`.")
+
+    mc = McSolver(H, c_ops, options=options)
+    result = mc.run(state, tlist=tlist, ntraj=ntraj, e_ops=e_ops,
+                    seed=seeds, target_tol=target_tol, timeout=timeout)
+    return result
+
+
+class _McTrajectorySolver(_TrajectorySolver):
+    """
+    Solver for one mcsolve trajectory. Created by a :class:`McSolver`.
+    """
+    name = "mcsolve"
+    _avail_integrators = {}
+
+    def __init__(self, rhs, c_ops, n_ops, *, options=None, _prepare=False):
+        self._c_ops = c_ops
+        self._n_ops = n_ops
+        super().__init__(rhs, options=options, _prepare=_prepare)
+
+    def _run(self, seed, state, tlist, e_ops):
+        self.collapses = []
+        self.generator = self.get_generator(seed)
```

Comment:
I think this is also done in super. I prefer to leave the handling of the seed to super such that the resposibilities of the classes are well separated.